src/components/DownloadModal/index.tsx

```diff
@@ -0,0 +1,414 @@
+import React, { Component } from 'react';
+// tslint:disable-next-line
+import Modal from 'react-modal';
+import { Colors, Styles, blueGradientBgImg } from '../../constants';
+import { TaskButton } from './TaskButton';
+import _ from 'lodash';
+// tslint:disable: no-unsafe-any no-any;
+/**
+ * Properties for DownloadModal
+ * @export
+ * @interface DownloadModalProps
+ */
+export interface DownloadModalProps {
+  taskModels?: string[];
+  isOpen: boolean;
+}
+/**
+ * State for DownloadModal
+ * @export
+ * @interface DownloadModalProps
+ */
+export interface DownloadModalState {
+  showModal: boolean;
+  showHide: string;
+  showMultiSelect: string;
+  taskModelButtons: JSX.Element[];
```

Comment:
Rather than storing an array of taskModelButtons here we should look at what information we need at the parent level to tell the child component how to render. We should only store the information that we will need to change in the child component and then pass it down through props.

Reply:
This may require creating some new interfaces to represent that data.